website/api/zkvm/developer-guide/optimization.md

```diff
@@ -0,0 +1,441 @@
+# Guest Optimization Guide
+
+RISC Zero’s zkVM is designed and built to act like a physical CPU.
+We did this so you could use general purpose languages (e.g. Rust) and general purpose tools (e.g. Cargo, LLVM). **Generally, you can apply general purpose optimization techniques too!**
+
+In this guide, we’ll cover optimization of [zkVM guest programs].
+We’ll discuss when you can apply common techniques from other contexts, and when you need a new approach.
+
+[zkVM guest programs]: ./guest-code-101
+
+## Background
+
+### What is the zkVM, really?
+
+**The zkVM is essentially a CPU.**
+
+In particular, it is an implementation of the [RISC-V architecture] (specifically riscv32im), similar to how your laptop’s CPU may be an implementation of an [x86] or [ARM] architecture.
+This biggest difference is that the zkVM is implemented with [arithmetic circuits], in software, instead of circuitry made from silicon and copper.
+
+[RISC-V architecture]: /reference-docs/about-risc-v
+[x86]: https://en.wikipedia.org/wiki/X86
+[ARM]: https://en.wikipedia.org/wiki/ARM_architecture_family
+[arithmetic circuits]: /reference-docs/about-arithmetic-circuits
+
+### What is a “cycle”?
```

Comment:
Maybe we should add a link to this section as a "see also" in the "clock cycle" entry on the Terminology page.

Reply:
I'm not sure that would improve clarity. I worry the reader would click the link, and be confused to see the same definition repeated. (I also don't want to only have a link, and no inline definition, because this is a really core concept and I don't want them to miss it)

Reply:
I was proposing a link in the other direction: from Terminology to this article. In that direction, it seems like having this long performance article is giving a "why you care" context to the cycle definition.

Reply:
Ah yes! I went to add it but realized that the issue we face is that this doc will land on the `next` API version. I don't think we want to link from Terminology to docs under `next`, so I think we should add that link when the page is under a released version.

Added the line as a comment in 84b3c95d3aa9038ed5739d9c1f28eb80d1d9cdde